File: 2Star AdvLvl/CHEFPATH.py
def magicalPath():
    a, b = map(int, input().split())
    if (a == 1 and b == 2) or (a == 2 and b == 1):
        print("Yes")
    elif a == 1 or b == 1:
        print("No")
    elif a % 2 != 0 and b % 2 != 0:
        print("No")
    else:
        print("Yes")

t = int(input())
for _ in range(t):
    magicalPath()

File: 2Star AdvLvl/CHEFPATH.java
import java.util.Scanner;

public class Main {
    public static void magicalPath() {
        Scanner scanner = new Scanner(System.in);
        long a = scanner.nextLong();
        long b = scanner.nextLong();
        if ((a == 1 && b == 2) || (a == 2 && b == 1))
            System.out.println("Yes");
        else if (a == 1 || b == 1)
            System.out.println("No");
        else if (a % 2 != 0 && b % 2 != 0)
            System.out.println("No");
        else
            System.out.println("Yes");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while (t-- > 0) {
            magicalPath();
        }
    }
}
